### PR TITLE
fix(PPDSC-2025): Tabbing into a checkbox breaks form validation

### PR DESCRIPTION
**Pull request by pdimova**
PPDSC-2025

**Before:**
https://user-images.githubusercontent.com/16742525/158382208-e7745b75-837d-428b-b7a4-4af0e59fa5fc.mp4

**After:**
https://user-images.githubusercontent.com/16742525/158382250-bb2e3bb9-f38b-4101-a087-b3c7ecfc155a.mp4

**I have done:**
 - [ ] Written unit tests against changes
 - [ ] Written accessibility tests against the component
 - [ ] Written functional tests against the component and/or NewsKit site
 - [ ] Updated relevant documentation
 - [ ] Updated the release notes for the [next release version](https://nidigitalsolutions.jira.com/wiki/spaces/NPP/pages/2349695021/Release%2BNotes%2BCode%2B-%2BWeb%2BDocumentation)
 - [ ] Discussed with QA and/or design about any visual changes

**I have tested manually:**
 - [ ] Renders correctly on IE11
 - [ ] The feature's functionality is working as expected on other browsers such as Safari, Firefox and IE11
 - [ ] There are no accessibility errors reported by WAVE
 - [ ] The screen reader reads the expected text for the tabbable elements.
 - [ ] There are no new errors in the browser console coming from this PR.
 - [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
 - [ ] The Playground feature is working as expected

**Before**

**After:**

**Who should review this PR**

**How to test**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on: 
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2025-tabbing-into-a-checkbox-breaks-form-validation 
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2025-tabbing-into-a-checkbox-breaks-form-validation/storybook

**Review by Xin00163 — APPROVED** [review]

File: src/form/__tests__/form.stories.tsx [hunk 1]
```diff
@@ -365,3 +365,41 @@ export const StoryFormWithCustomStyles = () => {
   );
 };
 StoryFormWithCustomStyles.storyName = 'form-with-custom-styles';
+
+export const StoryRenderBug = () => (
```
Comment (Vanals) [inline src/form/__tests__/form.stories.tsx:369]: Can we name it like the story? I think will lose sense at some point; I would not name it after the render team

**Comment by Vanals** [comment]
Wonder if worth double-checking with designers if this is expected. We can make another ticket if needed but just felt weird that once I check the check box it is still red. 

Once I submit the checkbox becomes invalid, if I go there and I check I guess I would expect it to be immediately green without having to tab out 🤔 . Not sure felt weird.

https://user-images.githubusercontent.com/25549355/159035668-f460ca46-f915-43a9-bedb-36da3dbf48f9.mp4

**Review by mutebg — APPROVED** [review]

**Comment by pdimova** [comment]
> Wonder if worth double-checking with designers if this is expected. Scenario: Leave the checkbox unchecked, submit, check and the checkbox once checked it still looks invalid. I maybe would expect it to be immediately green without having to tab out 🤔 .
> 
>  checkbox-behaviour.mp4

In this scenario the validation and re-validation mode is on blur event, so submitting the form won't update the valid states. only tabbing in and out will do so

lets discuss this when you're online

**Review by mutebg — APPROVED** [review]

**Pull request merged**